src/components/InventoryV2/Search.tsx

```diff
@@ -1,38 +1,85 @@
 import strings from 'src/strings';
-import InventoryFilters, { InventoryFiltersType } from './InventoryFiltersPopover';
-import { getNurseryName } from './FilterUtils';
 import { useOrganization } from 'src/providers/hooks';
-import { PillList } from '@terraware/web-components';
-import theme from '../../theme';
+import theme from 'src/theme';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { selectSpecies } from 'src/redux/features/species/speciesSelectors';
+import { requestSpecies } from 'src/redux/features/species/speciesThunks';
+import { getNurseryName } from './FilterUtils';
+import InventoryFilters, { InventoryFiltersType } from './InventoryFiltersPopover';
+import { OriginPage } from './InventoryBatch';
 
 interface SearchProps {
   searchValue: string;
   onSearch: (value: string) => void;
   filters: InventoryFiltersType;
   setFilters: React.Dispatch<React.SetStateAction<InventoryFiltersType>>;
+  origin?: OriginPage;
 }
 
-export default function Search(props: SearchProps): JSX.Element {
-  const { selectedOrganization } = useOrganization();
+type PillListItemWithEmptyValue = PillListItem<string> & { emptyValue: unknown };
+
+export default function Search(props: SearchProps): JSX.Element | null {
   const { searchValue, onSearch, filters, setFilters } = props;
-  const [filterPillData, setFilterPillData] = useState<PillListItem<string>[]>();
+  const origin = props.origin || 'Species';
+
+  const { selectedOrganization } = useOrganization();
+  const dispatch = useAppDispatch();
+
+  const species = useAppSelector(selectSpecies);
+
+  const [filterPillData, setFilterPillData] = useState<PillListItemWithEmptyValue[]>([]);
+
+  const getSpeciesName = useCallback(
+    (speciesId: number) => (species || []).find((s) => s.id === speciesId)?.commonName,
+    [species]
+  );
 
   useEffect(() => {
-    let data: PillListItem<string>[] = [];
+    let data: PillListItemWithEmptyValue[] = [];
     if ((filters.facilityIds?.length ?? 0) > 0) {
       data = [
         {
-          id: 'filterPillData',
+          id: 'facilityIds',
           label: strings.NURSERY,
-          value: filters.facilityIds?.map((id) => getNurseryName(id, selectedOrganization))?.join(', ') ?? '',
+          value: filters.facilityIds?.map((id) => getNurseryName(id, selectedOrganization)).join(', ') ?? '',
+          emptyValue: [],
+        },
+      ];
+    } else if ((filters.speciesIds?.length ?? 0) > 0) {
+      data = [
+        {
+          id: 'speciesIds',
+          label: strings.SPECIES,
+          value: filters.speciesIds?.map(getSpeciesName).join(', ') ?? '',
+          emptyValue: [],
         },
       ];
     }
+
     setFilterPillData(data);
-  }, [selectedOrganization, filters.facilityIds]);
+  }, [selectedOrganization, filters.facilityIds, filters.speciesIds, getSpeciesName]);
+
+  useEffect(() => {
+    if (origin === 'Nursery') {
+      void dispatch(requestSpecies(selectedOrganization.id));
```

Comment:
Why do we need that "void" before the dispatch?

Reply:
Because we are explicitly ignoring the promise returned. It just indicates that we know it is a promise, but in this case we are not doing anything with the response.

This is what shows up in my IDE if I leave it out
![SCR-20231117-krlp](https://github.com/terraware/terraware-web/assets/9541815/ef4a3503-5180-461d-b177-e40bae83f297)

Reply:
I don't get that warning in my IDE, so thanks for the explanation!